### PR TITLE
Improve build system configuration

### PR DESCRIPTION
**Pull request by jonkoops**
Changes:
- WebPack config files are now written in TypeScript
- Polyfills are added automatically through babel-env
- Dependencies written in ES Modules now supported
- Seperate build for legacy platforms such as Internet Explorer (differential builds)
- Modern build shipping with ES2015 code for modern platforms

### CODE REVIEW

File: src/app/AppBody.jsx [hunk 1]
```diff
@@ -12,21 +12,41 @@ import LoadingIndicator from '../shared/components/loading-indicator/LoadingIndi
...
-const HomePage = React.lazy(() => import('./pages/HomePage'))
-const ActualityContainer = React.lazy(() => import('./containers/ActualityContainer'))
+const HomePage = React.lazy(() => import(/* webpackChunkName: "HomePage" */ './pages/HomePage'))
+const ActualityContainer = React.lazy(() =>
+  import(/* webpackChunkName: "ActualityContainer" */ './containers/ActualityContainer'),
+)
 const DatasetDetailContainer = React.lazy(() =>
-  import('./containers/DatasetDetailContainer/DatasetDetailContainer'),
+  import(
+    /* webpackChunkName: "DatasetDetailContainer" */ './containers/DatasetDetailContainer/DatasetDetailContainer'
```
Comment (chrisvanmook) [inline src/app/AppBody.jsx:21]: Are we chunking all pages?

File: webpack.common.ts [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+          },
+        },
+        {
+          test: /\.svg$/,
+          use: [
+            {
+              loader: '@svgr/webpack',
+              options: {
+                svgo: options.mode === 'production',
+                svgoConfig: {
```
Comment (chrisvanmook) [inline webpack.common.ts:141]: same config as above, any way we can put this in a variable?

File: webpack.prod.ts [hunk 3]
```diff
@@ -1,32 +1,39 @@
...
-      'leaflet|leaflet-draw|leaflet-rotatedmarker|leaflet.markercluster|leaflet.nontiledlayer|proj4|proj4leaflet',
-    DATAPUNT: '@datapunt',
-    STYLED: 'styled-components|polished|style-loader|css-loader|sass-loader|postcss-loader',
-    PANORAMA: 'marzipano',
-    POLYFILL: '@babel/polyfill|objectFitPolyfill|core-js',
-    ANGULAR: 'angular|angular-aria|angular-sanitize|react-angular',
-    REACT:
-      'react|react-dom|redux-first-router|redux-first-router-link|redux-first-router-restore-scroll|reselect|redux|@?redux-saga|react-redux|react-helmet|prop-types',
-  }
+const CHUNKS = {
```
Comment (chrisvanmook) [inline webpack.prod.ts:6]: Isn't this gonna clash with the lazy imports?

File: src/app/AppBody.jsx (hunk 1 above)
Comment (jonkoops) [reply src/app/AppBody.jsx:21]: Most of them seem to be split into seperate lazy-loaded chunks. There are some optimizations that we can still do looking at the bundle, but I consider that out of scope for this PR. I have however created an issue on our backlog so we can take a look at this later.

File: webpack.prod.ts (hunk 3 above)
Comment (jonkoops) [reply webpack.prod.ts:6]: It shouldn't. The chunks should load in whenever a module that requires them is loaded.

File: webpack.common.ts (hunk 2 above)
Comment (jonkoops) [reply webpack.common.ts:141]: We could, but I don't see the advantage of it as this config is only used once.

Comment (chrisvanmook) [reply webpack.common.ts:141]: It's used here as well, right? https://github.com/Amsterdam/atlas/pull/2641/files#diff-ff4fc69cd4b6201389c9738391bd9e0eR220

Comment (jonkoops) [reply webpack.common.ts]: Oh, yeah nice catch. I merged the two definitions.

**Pull request merged**